activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -526,8 +526,25 @@ def default_index_type?(index) # :nodoc:
         index.using == :btree || super
       end
 
+      def insert_fixtures(*)
+        without_sql_mode('NO_AUTO_VALUE_ON_ZERO') { super }
```

Comment:
In bulk insert, we rely on queries like `INSERT INTO users (id, name) VALUES (DEFAULT, 'David'), (DEFAULT, 'Rafael')`. This kind of query doesn't work with `NO_AUTO_VALUE_ON_ZERO` flag.

On MySQL adapter, we [enable](https://github.com/rails/rails/blob/87663ea14034a9a659a7dd617eb575df3fb9bb62/activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb#L756) flag option by default. The solution is to temporary disable this flag to insert the fixtures.